### PR TITLE
Bugfix/2935 lakectl bug on not found html

### PR DESCRIPTION
**Pull request by idanovo**

### CODE REVIEW

**Comment by BinyaminSchein** [comment]
Loving thisssss

File: cmd/lakectl/cmd/annotate.go [hunk 1]
```diff
@@ -31,29 +33,29 @@ var annotateCmd = &cobra.Command{
...
 				res, err := client.LogCommitsWithResponse(context, pathURI.Repository, pathURI.Ref, logCommitsParams)
-				DieOnResponseError(res, err)
+				DieOnErrorOrUnexpectedStatusCode(resp, err, http.StatusOK)
 				data := objectCommitData{
 					Object: obj.Path,
 				}
 				if len(res.JSON200.Results) > 0 {
 					data.Commit = res.JSON200.Results[0]
-					data.CommitMessage = splitOnNewLine(stringTrimLen((res.JSON200.Results[0].Message), annotateMessageSize))
+					data.CommitMessage = splitOnNewLine(stringTrimLen(res.JSON200.Results[0].Message, annotateMessageSize))
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/annotate.go:58]: This seems unrelated.  Also I am not sure it does what is intended here (I am not sure what is intended here).  Maybe leave it to a separate PR?

File: cmd/lakectl/cmd/common_helpers.go [hunk 2]
```diff
@@ -206,6 +210,31 @@ func DieOnResponseError(response interface{}, err error) {
 		DieErr(retrievedErr)
 	}
 }
+func DieOnErrorOrUnexpectedStatusCode(response interface{}, err error, expectedStatusCode int) {
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/common_helpers.go:213]: Could we refactor parts of HTTPResponseAsError with this?  I think it copies some parts.

File: cmd/lakectl/cmd/doctor.go [hunk 3]
```diff
@@ -127,7 +128,7 @@ func ListRepositoriesAndAnalyze(ctx context.Context) error {
 	case resp.JSON401 != nil:
 		return &CredentialsError{msgOnErrCredential, resp.JSON401.Message}
 	// In case we get the "not found" HTML page (the status is 200 and not 404 in this case).
-	case resp.HTTPResponse != nil && resp.HTTPResponse.StatusCode == 200:
+	case resp.HTTPResponse != nil && resp.HTTPResponse.StatusCode == 302:
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/doctor.go:131]: The comment doesn't change?

File: docs/reference/commands.md [hunk 4]
```diff
@@ -1339,7 +1339,7 @@ lakectl branch-protect add <repo uri> <pattern> [flags]
 {:.no_toc}
 
 ```
-lakectl add lakefs://<repository> 'stable_*'
+lakectl branch-protect add lakefs://<repository> 'stable_*'
```
Comment (arielshaqed) [inline docs/reference/commands.md:1342]: These seem unrelated, and I'd rather the author reviewed this.

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Thanks!

1 res/resp mixup that has to be fixed, and some things don't belong in this PR I think.

File: cmd/lakectl/cmd/doctor.go (hunk 3 above)
Comment (idanovo) [reply cmd/lakectl/cmd/doctor.go:131]: Which comment? The output comment of the `doctor`?

File: cmd/lakectl/cmd/common_helpers.go (hunk 2 above)
Comment (idanovo) [reply cmd/lakectl/cmd/common_helpers.go:213]: I've tried to, but it seems to be kind of difficult because of the structure of `ResponseAsError` func. All we want in our case is to get the status code from the response when `ResponseAsError` checks and returns an error on its way to get it.  WDYT?

File: docs/reference/commands.md (hunk 4 above)
Comment (idanovo) [reply docs/reference/commands.md:1342]: Opened #2976

File: cmd/lakectl/cmd/annotate.go [hunk 5]
```diff
@@ -31,29 +33,29 @@ var annotateCmd = &cobra.Command{
...
-			DieOnResponseError(resp, err)
+			DieOnErrorOrUnexpectedStatusCode(resp, err, http.StatusOK)
 			for _, obj := range resp.JSON200.Results {
 				logCommitsParams := &api.LogCommitsParams{
 					Amount:  api.PaginationAmountPtr(1),
 					Objects: &[]string{obj.Path},
 				}
 				res, err := client.LogCommitsWithResponse(context, pathURI.Repository, pathURI.Ref, logCommitsParams)
-				DieOnResponseError(res, err)
+				DieOnErrorOrUnexpectedStatusCode(resp, err, http.StatusOK)
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply cmd/lakectl/cmd/annotate.go:52]: `tag.go` still contains a variable `res`:
```sh
[ariels@fedora lakeFS]$ git grep -w res cmd/
cmd/lakectl/cmd/tag.go:                 res, err := client.GetCommitWithResponse(ctx, tagURI.Repository, commitRef)
cmd/lakectl/cmd/tag.go:                 DieOnErrorOrUnexpectedStatusCode(res, err, http.StatusOK)
```

This inconsistency is precisely what got us into trouble on this line.  So could you fix that one too, please, even though it is not strictly related to this PR?

File: cmd/lakectl/cmd/tag.go [hunk 6]
```diff
@@ -78,8 +78,8 @@ var tagCreateCmd = &cobra.Command{
 		force, _ := cmd.Flags().GetBool("force")
 		if force {
 			// checking validity of the commitRef before deleting the old one
-			resp, err := client.GetCommitWithResponse(ctx, tagURI.Repository, commitRef)
-			DieOnErrorOrUnexpectedStatusCode(resp, err, http.StatusOK)
+			res, err := client.GetCommitWithResponse(ctx, tagURI.Repository, commitRef)
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/tag.go:81]: Sorry for being unclear: I really liked this change, it's part of the res -> resp fix.  It caused an issue with this PR, so I think we have no choice but to make this unrelated change here.  (Technically we could have a preview PR that just fixed this, and then your PR.  Too much effort for too little gain probably.)

**Review by arielshaqed — APPROVED** [review]
Thanks!

File: cmd/lakectl/cmd/tag.go (hunk 6 above)
Comment (idanovo) [reply cmd/lakectl/cmd/tag.go:81]: not sure how we can fix this in this case. We are trying to assign `resp` in l.84 with a different type. Maybe change `res` here to be `_resp`?

File: cmd/lakectl/cmd/abuse.go [hunk 7]
```diff
@@ -119,7 +120,7 @@ var abuseRandomWritesCmd = &cobra.Command{
 
 		client := getClient()
 		resp, err := client.GetRepositoryWithResponse(cmd.Context(), u.Repository)
-		DieOnResponseError(resp, err)
+		DieOnErrorOrUnexpectedStatusCode(resp, err, http.StatusOK)
```
Comment (nopcoder) [inline cmd/lakectl/cmd/abuse.go:123]: suggestion - pick a shorter name

File: cmd/lakectl/cmd/annotate.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply cmd/lakectl/cmd/annotate.go:52]: My 2 cents - use different names so we will not verify or check the wrong response by mistake

File: cmd/lakectl/cmd/branch_protect.go [hunk 8]
```diff
@@ -1,7 +1,10 @@
 package cmd
 
 import (
+	"net/http"
+
 	"github.com/spf13/cobra"
+
```
Comment (nopcoder) [inline cmd/lakectl/cmd/branch_protect.go:7]: remove whitespace

File: cmd/lakectl/cmd/common_helpers.go [hunk 9]
```diff
@@ -41,6 +43,8 @@ const (
 const internalPageSize = 1000          // when retreiving all records, use this page size under the hood
 const defaultAmountArgumentValue = 100 // when no amount is specified, use this value for the argument
 
+const defaultResponseOnSwaggerClient = http.StatusNoContent
```
Comment (nopcoder) [inline cmd/lakectl/cmd/common_helpers.go:46]: Prefer to use http.StatusNoContent when needed - the code will be readable without this indirection

File: cmd/lakectl/cmd/root.go [hunk 10]
```diff
@@ -108,9 +109,18 @@ func getClient() *api.ClientWithResponses {
 	if u.Path == "" || u.Path == "/" {
 		serverEndpoint = strings.TrimRight(serverEndpoint, "/") + api.BaseURL
 	}
+	httpClient := &http.Client{
+
```
Comment (nopcoder) [inline cmd/lakectl/cmd/root.go:113]: remove whitespace

File: pkg/api/ui_handler.go [hunk 11]
```diff
@@ -80,6 +80,8 @@ func NewHandlerWithDefault(root http.FileSystem, handler http.Handler, defaultPa
 		}
 		_, err := root.Open(path.Clean(urlPath))
 		if err != nil && os.IsNotExist(err) {
+			fmt.Println("redirecting...")
```
Comment (nopcoder) [inline pkg/api/ui_handler.go:83]: remove println

File: pkg/api/ui_handler.go [hunk 12]
```diff
@@ -80,6 +80,8 @@ func NewHandlerWithDefault(root http.FileSystem, handler http.Handler, defaultPa
...
+			http.Redirect(w, r, defaultPath, http.StatusFound)
```
Comment (nopcoder) [inline pkg/api/ui_handler.go:84]: As far as I remember, you need to return after redirect and not continue the code flow

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: cmd/lakectl/cmd/abuse.go (hunk 7 above)
Comment (idanovo) [reply cmd/lakectl/cmd/abuse.go:123]: DieOnUnexpectedResponse?

Comment (nopcoder) [reply cmd/lakectl/cmd/abuse.go:122]: Sorry, no good suggestion

**Review by nopcoder — APPROVED** [review]
Just address other comments
